pkg/deployment/inventory.go

```diff
@@ -99,40 +97,29 @@ func GenerateRoleInventory(ctx context.Context, helper *helper.Helper,
 		}
 	}
 
+	invData, err := inventory.MarshalYAML()
+	if err != nil {
+		utils.LogErrorForObject(helper, err, "Could not parse Role inventory", instance)
+		return "", err
+	}
+	cmData := map[string]string{
+		"inventory": string(invData),
+		"network":   string(instance.Spec.NodeTemplate.NetworkConfig.Template),
+	}
 	configMapName := fmt.Sprintf("dataplanerole-%s", instance.Name)
-	cm := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      configMapName,
-			Namespace: instance.Namespace,
-			Labels:    instance.ObjectMeta.Labels,
+	cms := []utils.Template{
+		// ConfigMap
```

Comment:
Not sure of the purpose of this comment? Should it be more explicit about which configmap is being created? Is the comment needed at all?

Reply:
Yes there is a puprose. utils.Template can be used for configmaps and secrets and this says it's a configmap[1].

[1] https://github.com/openstack-k8s-operators/infra-operator/blob/main/controllers/memcached/memcached_controller.go#L248